src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -36,8 +33,26 @@ private static string[] ParseMultiValue(string value)
                 }
                 else if ((value[i] == ',') && !inquote)
                 {
-                    singleValue = new string(vp, 0, chIndex);
-                    tempStringCollection.Add(singleValue.Trim());
+                    int start, end;
+
+                    for (start = 0; start < chIndex; ++start)
+                    {
+                        if (!char.IsWhiteSpace(vp[start]))
+                        {
+                            break;
+                        }
+                    }
+
+                    for (end = chIndex - 1; end >= start; ++end)
```

Comment:
This is broken; it needs to be ```--end``` rather than ```++end```.  And the resulting fact that the PR passed implies we don't have any tests covering the case of whitespace at the end of the string?  Can you add some more tests to ensure this is appropriately exercised?